dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/program/notification/DefaultProgramNotificationTemplateService.java

```diff
@@ -87,4 +98,35 @@ public void delete( ProgramNotificationTemplate programNotificationTemplate )
     {
         return store.getProgramNotificationByTriggerType( trigger );
     }
+
+    @Override
+    @Transactional( readOnly = true )
+    public boolean isProgramLinkedToWebHookNotification( Program program )
+    {
+        return programWebHookNotificationCache
```

Comment:
This cache can have at most all the program in the system.
Is 1000 as size limit too much?

Reply:
The number of program in a system is typically less than 100. I think 600 is the largest I have seen (just for context).

Reply:
setting it to 500